ddtrace/internal/datadog/profiling/cmake/AnalysisFunc.cmake

```diff
@@ -46,11 +43,19 @@ function(add_ddup_config target)
             -Wl,--exclude-libs,ALL)
     endif()
 
-    # If we can IPO, then do so
+    # If we can IPO, then do so.
     check_ipo_supported(RESULT result)
 
     if(result)
-        set_property(TARGET ${target} PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
+        if(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+            # When using AppleClang, explicitly use thin LTO to match Rust's thin LTO strategy. And set the object path
+            # for debug symbols.
```

Comment:
I'm a little confused by this. Why is this change necessary for this PR? Does link-time optimization affect debug symbols?

Reply:
Yeah, I think it makes sense to me now, thanks!